azure-functions/acbs-function/mappings/facility/facility-covenant-amend.js

```diff
@@ -0,0 +1,25 @@
+import { to2Decimals } from '../../helpers/currency';
+
+export const facilityCovenantAmend = (amendment) => {
+  try {
```

Comment:
I'm really not sure I like this pattern, I think it has a lot of potential for error and lacks a lot of validation that I'd expect to see. I'm worried it can fail silently and pass things we dont want onwards to ACBS...

I don't think it's particularly extendable, and I think it adds a fair amount of maintainence in regards to testing etc. I think the benifits of adding a validation layer and a mapping layer, using zod in this case, made a lot of sense as in addition to reusable complex validation rules, we could add reusable transformation patterns and validation, something we do not currently have.

But I understand this is an existing pattern, so happy to stick with it as a result.